src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -441,12 +441,24 @@ private static int GetMediaTypeLength(string input, int startIndex, out MediaTyp
             return current - startIndex;
         }
 
-        private static int GetMediaTypeExpressionLength(string input, int startIndex, out string mediaType)
+        internal static int ParseMediaTypeExpresion<T, K>(
+            string input,
+            int startIndex,
+            out K mediaType,
```

Comment:
`out` parameters go to the end? Not sure if there's a guidance for that but I've never seen them appear in the middle of an argument list.